rmgpy/solver/liquid.pyx

```diff
@@ -754,5 +811,9 @@ cdef class LiquidReactor(ReactionSystem):
                         if ir[j, 2] != -1:
                             pd[ir[j, 2], ip[j, 2]] += deriv
 
+        if residence_time:
+            pd -= 1/residence_time * np.identity(num_core_species, np.float64)
+
```

Comment:
I think for semi-batch there should be an extra term to account for the `dilution effect`, since the Volume change. I haven't done a complete proving but the term is something like `(-v_in/V) * np.identity(num_core_species, np.float64)`. I could be wrong. let me know your thoughts

Reply:
Thanks for your review and comment! And yes, you're right. I will modify the code.